### PR TITLE
typst-preview: 0.9.2 -> 0.10.5

### LINKED ISSUE #280865: Update request: typst-preview 0.9.2 → 0.10.5
- Package name: typst-preview
- Latest released version: 0.10.5

- Current version on the unstable channel: 0.9.2
- Current version on the stable/release channel: 0.9.1

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

Note: I attempted to update it myself, but the new version was restructured in a way I don't know how to translate to `mkYarnPackage`.

**Notify maintainers**

@berberman

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by berberman**
## Description of changes

Closes #280865

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ty/typst-preview/package.nix [hunk 1]
```diff
@@ -1,41 +1,35 @@
...
-, fetchYarnDeps
-, pkg-config
-, libgit2
-, openssl
-, zlib
-, stdenv
-, darwin
-}:
+{ lib, rustPlatform, fetchFromGitHub, mkYarnPackage, fetchYarnDeps, pkg-config
+, libgit2, openssl, zlib, stdenv, darwin }:
```
Comment (drupol) [inline pkgs/by-name/ty/typst-preview/package.nix:2]: Can we keep one line per dependency please?

File: pkgs/by-name/ty/typst-preview/package.nix [hunk 2]
```diff
@@ -50,29 +44,46 @@ let
...
-    pkg-config
-  ];
+  nativeBuildInputs = [ pkg-config ];
 
-  buildInputs = [
-    libgit2
-    openssl
-    zlib
-  ] ++ lib.optionals stdenv.isDarwin [
+  buildInputs = [ libgit2 openssl zlib ] ++ lib.optionals stdenv.isDarwin [
```
Comment (drupol) [inline pkgs/by-name/ty/typst-preview/package.nix:86]: Can you please revert this change?

File: pkgs/by-name/ty/typst-preview/package.nix [hunk 3]
```diff
@@ -50,29 +44,46 @@ let
...
-      "typst-ts-compiler-0.4.0-rc9" = "sha256-NVmbAodDRJBJlGGDRjaEcTHGoCeN4hNjIynIDKqvNbM=";
+      "typst-0.10.0" = "sha256-/Oy4KigXu1E/S9myd+eigqlNvk5x+Ld9gTL9dtpoyqk=";
+      "typst-ts-compiler-0.4.2-rc5" = "sha256-fhwTaAK19Nb7AKNJ9QBZgK1MO7g7s5AdSDqaBjLxT3w=";
     };
   };
 
-  nativeBuildInputs = [
-    pkg-config
-  ];
+  nativeBuildInputs = [ pkg-config ];
```
Comment (drupol) [inline pkgs/by-name/ty/typst-preview/package.nix:84]: Can you please revert this change?

File: pkgs/by-name/ty/typst-preview/package.nix [hunk 4]
```diff
@@ -1,41 +1,35 @@
...
-    packageJSON = ./package.json;
+    packageJSON = ./frontend.json;
 
     offlineCache = fetchYarnDeps {
       yarnLock = "${frontendSrc}/yarn.lock";
-      hash = "sha256-7a7/UOfau84nLIAKj6Tn9rTUmeBJ7rYDFAdr55ZDLgA=";
+      hash = "sha256-6e3UNd8gIBnTtllpo/1AC1XzeZ88rdUiechoQfo5V1Y=";
     };
 
+    packageResolutions = { typst-dom = dom; };
```
Comment (drupol) [inline pkgs/by-name/ty/typst-preview/package.nix:31]: Can we define `dom` just before using it? It's easier to read and understand.

**Review by drupol — CHANGES_REQUESTED** [review]
Hello,

Thanks for this massive update! I left some comments, let me know if this is clear enough.

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
@GrahamcOfBorg build typst-preview

**Comment by wackbyte** [comment]
thanks!

**Pull request merged**